components/HomePage/Contact.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+
+import { useState } from 'react';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { Textarea } from '@/components/ui/textarea';
+import { useToast } from '@/hooks/use-toast';
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '@/components/ui/card';
+import { HandleContactUs } from '@/app/actions/rabbitmq';
+
+const formSchema = z.object({
+  email: z.string().email({ message: 'Invalid email address' }),
+  message: z.string().min(8, { message: 'Message must be at least 8 characters long' }),
+});
+
+type FormData = z.infer<typeof formSchema>;
+
+export default function ContactSection() {
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
```

Comment:
isSubmitting is already provided by react hook form. Use that instead of seperate state.